### ISSUE TITLE
proposal: prop table api to add columns to the prop rows 

### ISSUE THREAD

**Issue by atanasster** [storybookjs/storybook#9691]
api to allow addons to create additional columns in the prop tables, for example for editing the values for the story. 
such an addon would need access to the full list of props to identify if it will add columns (for example the story might have a list of controls/knobs - and if one of their names corresponds to the one of the names of the prop table, it will creat an additional column (with a title)

screenshot of use case:
![grab44](https://user-images.githubusercontent.com/6075606/73506037-4fc07f80-43a2-11ea-8884-4b6f024a9fbc.jpg)

**Comment by shilman** [comment]
I'm ok with this as long as it's simple. Alternatively we could just build it into the props block directly, since I don't know what else we'd want to add. @atanasster do you have ideas here? @patricklafrance ?

**Comment by atanasster** [comment]
it could be as simple as just an entry point into the parameters.options . 
I would propose to add a field `extra` to PrepDef.
The callback would be called by the Props component in Docs, so it can send the enhanced rows to the PropsTable component.

**Comment by shilman** [comment]
I don't really like the idea of adding a new API for a single use case, but I guess it decouples the prop table from controls. To riff on your suggestion, I think any configuration should be scoped under `docs` parameter, or maybe even a new `propTable` parameter, to go along with the refactor proposed in #9690. And that we should try not to pollute `parameters.options` with too much random stuff.

**Comment by atanasster** [comment]
It forces me currently to include the prop editors into the lib/components package, since they are using the form controls from it but also the prop tables are using them.
Just the architecture is getting abit overweight.

**Comment by shilman** [comment]
@atanasster I agree. Using dependency injection (effectively) is a great way to keep things decoupled. I'm not excited about introducing a lot of new packages, but it could be a great way to make sure things stay in their place and avoid spaghetti-ness. 😄

**Comment by patricklafrance** [comment]
I understand why you want to do this. However it might become a maintenance nightmare if people start to write a lot of addons that add columns.

I fear we might need to design for super small column because we don't know what user will come up with.

**Comment by atanasster** [comment]
Currently not a great situation either, i ended up cramping another 10-20 components into storybook/components and also its hard-coded to work only with the property controls. 

At the end it will be a compromise either way - i personally dislike hard-coding features - there are a few such examples with docs that i think long term will need to be taken out.
But i am also not a fan of extra work, so i put this out there and see the feedback.

**Comment by patricklafrance** [comment]
I agree that none of the choices are ideal. Maybe @domyen could offer some advices about how he see this from a design standpoint?

**Comment by atanasster** [comment]
The design as well, but this proposal is just for the architecture as storybook/components is getting a bit largish

**Comment by shilman** [comment]
I think we should do what’s right architecturally and not worry too much about the user adding too many columns. Let’s get it working with an extra column per @domyen ’s design and if we start to see great use cases with more columns we can do the extra work to handle that case then.

**Comment by anku1612** [comment]
Hi @atanasster 
How can we implement that to add extra column in bit.dev can u pls explain here it will be very benefical for me and other people as well

**Comment by valentinpalkovic** [comment]
The `ArgsTable` block this proposal targets was split in Storybook 8 into the read-only `<ArgTypes>` block and the editable `<Controls>` block — so "add an editable column" is now solved by composing `<Controls>` alongside `<ArgTypes>`. Addons can still enrich the data via `argTypesEnhancers`. Closing — no API for column injection is planned.
